apps/qubit/modules/settings/actions/diacriticsAction.class.php

```diff
@@ -0,0 +1,104 @@
+ */
+
+class SettingsDiacriticsAction extends SettingsEditAction
+{
+    // Arrays not allowed in class constants
+    public static $NAMES = [
+        'diacritics',
+    ];
+
+    public function earlyExecute()
+    {
+        parent::earlyExecute();
+
+        $this->updateMessage = $this->i18n->__('Diacritics settings saved.');
+
+        $this->settingDefaults = [
+            'diacritics' => QubitSetting::getByName('diacritics')->getValue(['sourceCulture' => true]),
+        ];
+    }
+
+    public function execute($request)
+    {
+        DefaultEditAction::execute($request);
+
+        if ($request->isMethod('post')) {
+            $this->form->bind($request->getPostParameters(), $request->getFiles());
+
+            if ($this->form->isValid()) {
+                // Check that a diacritics mapping yaml file is uploaded when enabling
+                if (0 === intval($request->getPostParameters()['diacritics']) || (1 === intval($request->getPostParameters()['diacritics']) && 0 !== $request->getFiles()['mappings']['size'])) {
+                    $this->processForm($request);
+                    $this->uploadDiacritics($request);
+
+                    $this->getUser()->setFlash('notice', $this->context->i18n->__('Diacritics setting updated. Rebuild the search index for the changes to be applied.'));
+                } else {
+                    $this->getUser()->setFlash('error', $this->context->i18n->__('Error: Unable to upload diacritics mapping.'));
+                }
+
+                $this->redirect(['module' => 'settings', 'action' => 'diacritics']);
+            }
+        }
+    }
+
+    public function processForm($request)
+    {
+        sfConfig::set('app_diacritics', $request->getPostParameters()['diacritics']);
+        QubitSetting::findAndSave('diacritics', $request->getPostParameters()['diacritics'], ['sourceCulture' => true]);
+    }
+
+    public function uploadDiacritics($request)
+    {
+        $file = $request->getFiles('mappings');
+        $diacriticsMappingPath = sfConfig::get('sf_plugins_dir').DIRECTORY_SEPARATOR.'arElasticSearchPlugin'.DIRECTORY_SEPARATOR.'config'.DIRECTORY_SEPARATOR.'diacritics_mapping.yml';
+
+        if (1 === (int) $request->getPostParameters()['diacritics']) {
+            if ('application/x-yaml' === $request->getFiles()['mappings']['type']) {
+                // Move uploaded file to replace diacritics_mapping.yml.
+                try {
+                    $file = Qubit::moveUploadFile($file);
+                } catch (sfException $e) {
+                    $this->getUser()->setFlash('error', $e->getMessage());
+                }
+
+                // Replace /config/diacritics_mapping.yml with uploaded file
+                $uploadMappingPath = $file['tmp_name'];
+                $diacriticsMappings = file_get_contents($uploadMappingPath);
+                file_put_contents($diacriticsMappingPath, $diacriticsMappings);
+            }
```

Comment:
Might be good to put an else in this logic to flash an error at the user if file isn't "application/x-yaml".

Reply:
thank you for your patience with my work XD i also updated diacriticsSuccess.php to make the save button match those in BS5, but that changed the look on BS2 theme. any suggestions on making them match other save buttons in both themes?

Reply:
I'm not too up-to-speed on BS5 but I'll take a look!

Reply:
i added a success page to the BS5 theme plugin to resolve this issue!

Reply:
> Might be good to put an else in this logic to flash an error at the user if file isn't "application/x-yaml".

If setting the mine type in the sfValidatorFile, it will work like a normal form error.